vscode/src/definition.ts

```diff
@@ -23,10 +24,28 @@ class QSharpDefinitionProvider implements vscode.DefinitionProvider {
     );
     if (!definition) return null;
     const uri = vscode.Uri.parse(definition.source);
-    // We have to do this to map the position :(
-    const definitionPosition = (
-      await vscode.workspace.openTextDocument(uri)
-    ).positionAt(definition.offset);
+
+    const definitionPosition = (await openDocument(uri)).positionAt(
+      definition.offset,
+    );
     return new vscode.Location(uri, definitionPosition);
   }
 }
+
+async function openDocument(uri: vscode.Uri) {
```

Comment:
I'm wondering if this is going to need to be used more broadly across several different LS features like rename and FAR.

Reply:
Yes, this will probably be used for any language feature that could return spans from different files. I can move it to a common location.